src/SigSpec.CodeGeneration.TypeScript/SigSpecToTypeScriptGenerator.cs

```diff
@@ -27,28 +27,27 @@ public CodeArtifactCollection GenerateArtifacts(SigSpecDocument document)
             {
                 var hubModel = new HubModel(hub.Key, hub.Value, resolver);
                 var template = _settings.TypeScriptGeneratorSettings.TemplateFactory.CreateTemplate("TypeScript", "Hub", hubModel);
-                artifacts.Add(new CodeArtifact(hubModel.Name, CodeArtifactType.Class, CodeArtifactLanguage.TypeScript, template));
+                artifacts.Add(new CodeArtifact(hubModel.Name, CodeArtifactType.Class, CodeArtifactLanguage.TypeScript, CodeArtifactCategory.Client, template.Render()));
             }
 
             if (_settings.GenerateDtoTypes)
             {
                 var generator = new TypeScriptGenerator(document, _settings.TypeScriptGeneratorSettings, resolver);
                 var types = generator.GenerateTypes();
 
-                return new CodeArtifactCollection(artifacts.Concat(types.Artifacts), types.ExtensionCode);
+                return artifacts.Concat(types);
             }
             else
             {
-                var extensionCode = new TypeScriptExtensionCode(_settings.TypeScriptGeneratorSettings.ExtensionCode, _settings.TypeScriptGeneratorSettings.ExtendedClasses);
-                return new CodeArtifactCollection(artifacts, extensionCode);
```

Comment:
Is it really correct to remove extension code handling here?
Might be missing now...

Reply:
Ye i wasn't excatly sure what that code did, i have attempted to put the code back in but not sure i understand the api for the Extension Code.

And to answer your other question yes CodeArtifactCollection has been removed and is now just an IEnumerable of CodeArtifacts